standard/ERCs/erc-6900.md

```diff
@@ -471,72 +471,55 @@ An account can have the same validation module installed more than once.
 The entityId of a validation function installed on an account MUST be unique.
 Validation installation MAY be deferred until a later time, such as upon first use.
 
-#### Responsibilties of `StandardExecutor` and `ModuleExecutor`
-
-`StandardExecutor` functions are used for open-ended calls to external addresses.
-
-`ModuleExecutor` functions are specifically used by modules to request the account to execute with account's context. Explicit permissions are required for modules to use `ModuleExecutor`.
-
-The following behavior MUST be followed:
-
-- `StandardExecutor` can NOT call module execution functions and/or `ModuleExecutor`. This is guaranteed by checking whether the call's target implements the `IModule` interface via ERC-165 as required.
-- `StandardExecutor` can NOT be called by module execution functions and/or `ModuleExecutor`.
-- Module execution functions MUST NOT request access to `StandardExecutor`, they MAY request access to `ModuleExecutor`.
-
-#### Calls to `installModule`
-
-The function `installModule` accepts 3 parameters: the address of the module to install, the Keccak-256 hash of the module's manifest, ABI-encoded data to pass to the module's `onInstall` callback.
-
-The function MUST retrieve the module's manifest by calling `moduleManifest()` using `staticcall`.
-
-The function MUST perform the following preliminary checks:
+During validation installation, the account MUST correctly set flags and other fields based on the incoming data provided by users.
 
-- Revert if the module has already been installed on the modular account.
-- Revert if the module does not implement ERC-165 or does not support the `IModule` interface.
-- Revert if `manifestHash` does not match the computed Keccak-256 hash of the module's returned manifest. This prevents installation of modules that attempt to install a different module configuration than the one that was approved by the client.
-- Revert if any address in `dependencies` does not support the interface at its matching index in the manifest's `dependencyInterfaceIds`, or if the two array lengths do not match, or if any of the dependencies are not already installed on the modular account.
+- the account MUST install all pre validation hooks required by the user and call `onInstall` with the user-provided data on the hook module to initialize the states.
```

Comment:
A bit of a meta-note for all locations in the spec where we describe the requirement to call `onInstall`: Technically, our account currently doesn't perform this call if the data is empty. With respect to the specificiation, I think we could change the requirement to either be:
1. If data is provided to <parameter_name>, the account MUST call `onInstall` with the data.
2. The account SHOULD call `onInstall` with the data in <parameter_name>.

This pattern appears in each install/uninstall for validation/execution.

Wdyt?

Reply:
good point. Updated wording to user intent rather than specific var/param names.

Reply:
I think the 2nd is nicer, since it implies more clearly that hooks may not always be initialized with `onInstall`, as it's a user choice.